plugin/plugin.go

```diff
@@ -0,0 +1,288 @@
+
+		// Log groups
+		logrus.Infof("\nGroups:")
+		for _, group := range suite.Groups {
+			logrus.Infof("\n- Group: %s", group.Name)
+			for _, method := range group.Methods {
+				logrus.Infof("\n  - Method: %s | Class: %s | Signature: %s", method.Name, method.ClassName, method.Signature)
+			}
+		}
+
+		// Log test details
+		logrus.Infof("\nTest Details:")
+		for _, class := range suite.Classes {
+			for _, test := range class.Tests {
+				logrus.Infof("\n- Test: %s | Status: %s | Duration: %s ms", test.Name, test.Status, test.DurationMS)
+				if test.Status == "FAIL" && test.Exception != "" {
+					logrus.Infof("\n    Exception: %s", test.Exception)
+				}
+			}
+		}
+	}
+
+	results.Total = totalTests
+	results.Failures = totalFailures
+	results.Skipped = totalSkipped
+	results.DurationMS = totalDuration
+	return results
+}
+
+// validateThresholds validates test report thresholds based on aggregate results.
+func validateThresholds(results Results, args Args) error {
+
+	if args.FailureOnFailedTestConfig && results.Failures > 0 {
+		return errors.New("\nbuild marked as failed due to failed configuration methods as FailureOnFailedTestConfig is true")
+	}
+
+	switch args.ThresholdMode {
+	case 1: // Absolute thresholds
+		if err := validateAbsoluteThresholds(results, args); err != nil {
+			return errors.New("\nabsolute threshold validation failed: " + err.Error())
+		}
+
+		if strings.ToUpper(args.JobStatus) == "FAILED" {
+			if err := validateUnstableAbsoluteThresholds(results, args); err != nil {
+				return errors.New("\nfail absolute threshold validation failed: " + err.Error())
+			}
+		}
+	case 2: // Percentage thresholds
+		if err := validatePercentageThresholds(results, args); err != nil {
+			return errors.New("\npercentage threshold validation failed: " + err.Error())
+		}
+
+		if strings.ToUpper(args.JobStatus) == "FAILED" {
+			if err := validateUnstablePercentageThresholds(results, args); err != nil {
+				return errors.New("\nfail percentage threshold validation failed: " + err.Error())
+			}
+		}
+	default:
+		return fmt.Errorf("\ninvalid ThresholdMode: %d, expected 1 (absolute) or 2 (percentage)", args.ThresholdMode)
+	}
+	return nil
+}
+
+// validateAbsoluteThresholds checks absolute thresholds.
+func validateAbsoluteThresholds(results Results, args Args) error {
+	if args.FailedFails > 0 && results.Failures > args.FailedFails {
+		return fmt.Errorf("\nnumber of failed tests (%d) exceeded the failure threshold (%d)", results.Failures, args.FailedFails)
+	}
+	if args.FailedSkips > 0 && results.Skipped > args.FailedSkips {
+		return fmt.Errorf("\nnumber of skipped tests (%d) exceeded the skip threshold (%d)", results.Skipped, args.FailedSkips)
+	}
+	return nil
+}
+
+// validatePercentageThresholds checks percentage-based thresholds.
+func validatePercentageThresholds(results Results, args Args) error {
+	totalTests := results.Total
+	if totalTests == 0 {
+		return nil // No tests to validate
```

Comment:
```suggestion
	if totalTests == 0 {
	        logrus.Warn("No tests executed; skipping percentage-based threshold validation.")
		return nil
```

also add a test case to verify behavior when no tests are executed